jobs/ror/ProcessRorCsv.php

```diff
@@ -0,0 +1,287 @@
+     */
+    public bool $failOnTimeout = false;
+
+    use Batchable;
+
+    public function __construct(
+        protected string $pathCsv,
+        protected int $startRow,
+        protected int $endRow,
+        protected array $dataMapping,
+        protected array $dataMappingIndex,
+        protected string $noLocale,
+        protected string $temporaryTable,
+        protected string $scheduledTaskLogFilePath
+    )
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Get the middleware the job should pass through.
+     */
+    public function middleware()
+    {
+        return [
+            (new WithoutOverlapping("{$this->pathCsv}:{$this->startRow}-{$this->endRow}"))->expireAfter(60), // 1-minute lock
+        ];
+    }
+
+    public function handle()
+    {
+        try {
+            UpdateRorRegistryDataset::writeToExecutionLogFile(
+                "Importing batch chunk {$this->startRow}-{$this->endRow} starting",
+                $this->scheduledTaskLogFilePath,
+                ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_NOTICE
+            );
+
+            $this->createTemporaryTable();
+
+            $batchRows = [];
+            $currentRow = 0;
+            $isHeader = true;
+
+            if (($handle = fopen($this->pathCsv, 'r')) !== false) {
+                while (($rowCsv = fgetcsv($handle, null, ',', '"', '\\')) !== false) {
+                    if ($isHeader) {
+                        foreach ($this->dataMapping as $keyDB => $keyCsv) {
+                            $this->dataMappingIndex[$keyDB] = array_search($keyCsv, $rowCsv, true);
+                        }
+                        $isHeader = false;
+                        continue;
+                    }
+
+                    $currentRow++;
+                    if ($currentRow < $this->startRow) {
+                        continue;
+                    }
+                    if ($currentRow > $this->endRow) {
+                        break;
+                    }
+
+                    $batchRows[] = $this->processRow($rowCsv);
+                }
+                fclose($handle);
+
+                if (!empty($batchRows)) {
+                    $this->processBatch($batchRows);
+                }
+            }
+
+            $this->dropTemporaryTable();
+
+            UpdateRorRegistryDataset::writeToExecutionLogFile(
+                "Importing batch chunk {$this->startRow}-{$this->endRow} completed successfully",
+                $this->scheduledTaskLogFilePath,
+                ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_COMPLETED
+            );
+        } catch (Throwable $e) {
+            UpdateRorRegistryDataset::writeToExecutionLogFile(
+                "Importing batch chunk {$this->startRow}-{$this->endRow} failed: {$e->getMessage()}",
+                $this->scheduledTaskLogFilePath,
+                ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_ERROR
+            );
+
+            $this->dropTemporaryTable();
```

Comment:
This type of code that should be executed always can be moved to a `finally` section.